### ISSUE TITLE
json_extract func got error when json path has :  in key

### ISSUE THREAD

**Issue by rayshen92** [tjni/langgraph-checkpoint-mysql#9]
https://github.com/tjni/langgraph-checkpoint-mysql/blob/46a2599af9db0c36d6988e96f088e20dbeb61c22/langgraph/checkpoint/mysql/base.py#L61

```sql
json_extract(checkpoint, concat('$.channel_versions.',  channel))
```
this got error, when channel has :  char

fix method: add " around channel field
```sql
json_extract(checkpoint, concat('$.channel_versions.', '"', channel, '"'))
```

**Comment by asanger** [comment]
Can confirm, was running into this as well.

**Comment by f-makino-at-riracle** [comment]
I was just facing a similar problem (in MySQL 8.0.40 and 8.4.3) as well. 
Luckily, I found your some issues in here, applied the fix as a submodule, and it worked perfectly. 
How lucky I am!

Thanks!

**Comment by muyesh** [comment]
I was just facing a similar problem (in MySQL 8.0.40 and 8.4.3) as well. 
Luckily, I found this issue in here, applied the fix as a submodule, and it worked perfectly. 
How lucky I am!

Thanks @rayshen92 
And I am very grateful to author(@tjni ) who ported the library to MySQL.

**Comment by asanger** [comment]
@muyesh may I ask how you were able to create a submodule to modify the specific function? I was only able to get this to work by adding the full package source to my code and modifying it.

**Comment by muyesh** [comment]
I didn't provide enough explanation.
I just placed langgraph-checkpoint-mysql, which I had obtained via pip or poetry, as a git submodule and fixed the pointed-out issues.
```
$ git submodule add git@github.com:tjni/langgraph-checkpoint-mysql.git submodules/langraph-checkpoint-mysql
```

This is my directory structure.
```
$ tree -L 3
.
├── app
│   └── main.py
├── patches
│   └── langgraph-checkpoint-mysql.2024Nov20.patch
├── pyproject.toml
└── submodules (from github)
    └── langgraph-checkpoint-mysql
        ├── langgraph
        ├── LICENSE
        ├── Makefile
        ├── poetry.lock
        ├── pyproject.toml
        ├── README.md
        └── tests
```

./pyproject.toml
```
[tool.poetry.dependencies]
# langgraph-checkpoint-mysql = {extras = ["aiomysql"], version = "^2.0.2"}
langgraph-checkpoint-mysql = { extras = ["aiomysql"], path = "./submodules/langgraph-checkpoint-mysql" }
```

**Comment by tjni** [comment]
Thank you everyone for testing this package and providing fixes. I'm still working on adopting it in my organization, so I apologize for the glaring bugs. I've added tests for these bugs so they should not happen again.

I also released version 2.0.3, which should fix this.
